src/components/dashjs-api-control-element/dashjs-api-control-element.tsx

```diff
@@ -65,41 +48,66 @@ export class DashjsAPIControlElement {
       });
       toast.present();
     } else {
-      this.valueChanged.emit(this.functionValue);
+      this.valueChanged.emit(functionValue);
     }
   }
 
   render() {
-    const control = [];
+    let functionValue = [];
+    let control = [];
+
+    for (let i = 0; i < this.param.length; i++) {
+      const index = i;
+      switch (this.param[i].type) {
+        case 'string':
+          functionValue[index] = '';
+          break;
+        case 'number':
+          functionValue[index] = 0;
+          break;
+        case 'boolean':
+          functionValue[index] = false;
+          break;
+      }
+    }
+
+    let elem_values = LocalStorage.getKeyValueObject(STRING_API_ELEM);
+
+    if(elem_values) {
+      if(elem_values[this.name] != null) {
+        functionValue = this.param.length > 0 ? JSON.parse(elem_values[this.name]).length == this.param.length ? JSON.parse(elem_values[this.name]) : functionValue : elem_values[this.name];
+      }
+    }
+
     if (this.param.length > 0) {
       for (let i = 0; i < this.param.length; i++) {
         const index = i;
         switch (this.param[i].type) {
           case 'string':
             control.push(
-              <ion-input class="input-border" debounce={300} value={this.functionValue[index]} onIonChange={change => this.setValue(index, change.detail.value)}></ion-input>,
+              <ion-input class="input-border" debounce={300} value={functionValue[index]} onIonChange={change => { functionValue[index] = change.detail.value; this.setValue(functionValue)}}></ion-input>,
```

Comment:
maybe incorporate this into one function? because you wrote the same code 5 times